### PR TITLE
Create cluster on Azure for CarbonPlan

### PR DESCRIPTION
**Pull request by sgibson91**
This PR adds a terraform values file to create a cluster in Azure for the new carbonplan hub.

Here is a table that compares the node machines on the current AWS cluster and what we will deploy on the Azure cluster:

Screenshot 2021-11-15 at 14 07 08

<details> <summary> Output of 'terraform plan' </summary>

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_container_registry.container_registry will be created
  + resource "azurerm_container_registry" "container_registry" {
      + admin_enabled                 = true
      + admin_password                = (sensitive value)
      + admin_username                = (known after apply)
      + encryption                    = (known after apply)
      + georeplication_locations      = (known after apply)
      + georeplications               = (known after apply)
      + id                            = (known after apply)
      + location                      = "westus2"
      + login_server                  = (known after apply)
      + name                          = "2i2ccarbonplanhubregistry"
      + network_rule_set              = (known after apply)
      + public_network_access_enabled = true
      + resource_group_name           = "2i2c-carbonplan-cluster"
      + retention_policy              = (known after apply)
      + sku                           = "premium"
      + storage_account_id            = (known after apply)
      + trust_policy                  = (known after apply)
      + zone_redundancy_enabled       = false

      + identity {
          + identity_ids = (known after apply)
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = (known after apply)
        }
    }

  # azurerm_kubernetes_cluster.jupyterhub will be created
  + resource "azurerm_kubernetes_cluster" "jupyterhub" {
      + dns_prefix                          = "k8s"
      + fqdn                                = (known after apply)
      + id                                  = (known after apply)
      + kube_admin_config                   = (known after apply)
      + kube_admin_config_raw               = (sensitive value)
      + kube_config                         = (known after apply)
      + kube_config_raw                     = (sensitive value)
      + kubernetes_version                  = "1.20.7"
      + location                            = "westus2"
      + name                                = "hub-cluster"
      + node_resource_group                 = (known after apply)
      + portal_fqdn                         = (known after apply)
      + private_cluster_enabled             = (known after apply)
      + private_cluster_public_fqdn_enabled = false
      + private_dns_zone_id                 = (known after apply)
      + private_fqdn                        = (known after apply)
      + private_link_enabled                = (known after apply)
      + resource_group_name                 = "2i2c-carbonplan-cluster"
      + sku_tier                            = "Free"

      + addon_profile {
          + aci_connector_linux {
              + enabled     = (known after apply)
              + subnet_name = (known after apply)
            }

          + azure_policy {
              + enabled = (known after apply)
            }

          + http_application_routing {
              + enabled                            = (known after apply)
              + http_application_routing_zone_name = (known after apply)
            }

          + ingress_application_gateway {
              + effective_gateway_id                 = (known after apply)
              + enabled                              = (known after apply)
              + gateway_id                           = (known after apply)
              + gateway_name                         = (known after apply)
              + ingress_application_gateway_identity = (known after apply)
              + subnet_cidr                          = (known after apply)
              + subnet_id                            = (known after apply)
            }

          + kube_dashboard {
              + enabled = (known after apply)
            }

          + oms_agent {
              + enabled                    = (known after apply)
              + log_analytics_workspace_id = (known after apply)
              + oms_agent_identity         = (known after apply)
            }

          + open_service_mesh {
              + enabled = (known after apply)
            }
        }

      + auto_scaler_profile {
          + balance_similar_node_groups      = false
          + empty_bulk_delete_max            = (known after apply)
          + expander                         = (known after apply)
          + max_graceful_termination_sec     = (known after apply)
          + max_node_provisioning_time       = "15m"
          + max_unready_nodes                = 3
          + max_unready_percentage           = 45
          + new_pod_scale_up_delay           = (known after apply)
          + scale_down_delay_after_add       = (known after apply)
          + scale_down_delay_after_delete    = (known after apply)
          + scale_down_delay_after_failure   = (known after apply)
          + scale_down_unneeded              = (known after apply)
          + scale_down_unready               = (known after apply)
          + scale_down_utilization_threshold = (known after apply)
          + scan_interval                    = (known after apply)
          + skip_nodes_with_local_storage    = true
          + skip_nodes_with_system_pods      = true
        }

      + default_node_pool {
          + enable_auto_scaling  = true
          + kubelet_disk_type    = (known after apply)
          + max_count            = 10
          + max_pods             = (known after apply)
          + min_count            = 1
          + name                 = "core"
          + node_count           = 1
          + node_labels          = {
              + "hub.jupyter.org/node-purpose" = "core"
              + "k8s.dask.org/node-purpose"    = "core"
            }
          + orchestrator_version = "1.20.7"
          + os_disk_size_gb      = 40
          + os_disk_type         = "Managed"
          + os_sku               = (known after apply)
          + type                 = "VirtualMachineScaleSets"
          + ultra_ssd_enabled    = false
          + vm_size              = "Standard_E4s_v3"
          + vnet_subnet_id       = (known after apply)
        }

      + identity {
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = "SystemAssigned"
        }

      + kubelet_identity {
          + client_id                 = (known after apply)
          + object_id                 = (known after apply)
          + user_assigned_identity_id = (known after apply)
        }

      + linux_profile {
          + admin_username = "hub-admin"

          + ssh_key {
              + key_data = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCrtD6l/2S2dDT1OFLIUa46KAuzlUkckneQ4oyK5c2izT0nlBOxOcWACvnJ4rqntZVuhML7GYS35tQQnXyEXctTS2LnB7illl+oCT63WQzhhutfPLZSW3gJ+/CFVVp0VdHq4t1O8CQO+EgeV1cvuBre/neNZo5tcqXEsoV7tw/qw8XmD7Dk3yPz4NEFYZyZ4xQM7Qn2j9krZlRO2wCrosMFLo5Rv108PSDTHn3VtXEpJwemUO93D0ipJvCDE62M3vJxamqP4k5HLVn3Jun5KEvA7fINvgP3NSXkPVEEE4Prcqc1IJIzSxzygEN7xkZYsUEfJkybGXUWtUXdNjOEfLLlhYToMhdcP4jZluAjTtQmnJTr1sLegH+kbGIvI8MFIJAgXXLXS8/ubXHMozb+4gXRhyT8SMPwmgXGA+FtDduy5gnpPMSgmozQY17av5r25T6viWX5ivtL+n/CNJGf4npM1vKWcUNKiGyJrwzxQdpwJZ2uMP3EIdCYxOvqCQ+7UEM= sgibson@Athena.broadband"
            }
        }

      + network_profile {
          + dns_service_ip     = (known after apply)
          + docker_bridge_cidr = (known after apply)
          + load_balancer_sku  = "standard"
          + network_mode       = (known after apply)
          + network_plugin     = "kubenet"
          + network_policy     = "calico"
          + outbound_type      = "loadBalancer"
          + pod_cidr           = (known after apply)
          + service_cidr       = (known after apply)

          + load_balancer_profile {
              + effective_outbound_ips    = (known after apply)
              + idle_timeout_in_minutes   = (known after apply)
              + managed_outbound_ip_count = (known after apply)
              + outbound_ip_address_ids   = (known after apply)
              + outbound_ip_prefix_ids    = (known after apply)
              + outbound_ports_allocated  = (known after apply)
            }
        }

      + role_based_access_control {
          + enabled = (known after apply)

          + azure_active_directory {
              + admin_group_object_ids = (known after apply)
              + azure_rbac_enabled     = (known after apply)
              + client_app_id          = (known after apply)
              + managed                = (known after apply)
              + server_app_id          = (known after apply)
              + server_app_secret      = (sensitive value)
              + tenant_id              = (known after apply)
            }
        }

      + windows_profile {
          + admin_password = (sensitive value)
          + admin_username = (known after apply)
          + license        = (known after apply)
        }
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["huge"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nbhuge"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_E32s_v4"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_E32s_v4"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["large"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nblarge"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_E8s_v4"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_E8s_v4"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["medium"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nbmedium"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_E4s_v4"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_E4s_v4"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["small"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nbsmall"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_E2s_v4"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_E2s_v4"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["vhuge"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nbvhuge"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_M64s_v2"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_M64s_v2"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_kubernetes_cluster_node_pool.user_pool["vvhuge"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "user_pool" {
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = (known after apply)
      + max_count             = 20
      + max_pods              = (known after apply)
      + min_count             = 0
      + mode                  = "User"
      + name                  = "nbvvhuge"
      + node_count            = (known after apply)
      + node_labels           = {
          + "hub.jupyter.org/node-purpose"     = "user"
          + "k8s.dask.org/node-purpose"        = "scheduler"
          + "node.kubernetes.io/instance-type" = "Standard_M128s_v2"
        }
      + node_taints           = [
          + "hub.jupyter.org_dedicated=user:NoSchedule",
        ]
      + orchestrator_version  = "1.20.7"
      + os_disk_size_gb       = 200
      + os_disk_type          = "Managed"
      + os_sku                = (known after apply)
      + os_type               = "Linux"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_M128s_v2"
      + vnet_subnet_id        = (known after apply)
    }

  # azurerm_resource_group.jupyterhub will be created
  + resource "azurerm_resource_group" "jupyterhub" {
      + id       = (known after apply)
      + location = "westus2"
      + name     = "2i2c-carbonplan-cluster"
    }

  # azurerm_storage_account.homes will be created
  + resource "azurerm_storage_account" "homes" {
      + access_tier                      = (known after apply)
      + account_kind                     = "StorageV2"
      + account_replication_type         = "LRS"
      + account_tier                     = "Standard"
      + allow_blob_public_access         = false
      + enable_https_traffic_only        = true
      + id                               = (known after apply)
      + is_hns_enabled                   = false
      + large_file_share_enabled         = (known after apply)
      + location                         = "westus2"
      + min_tls_version                  = "TLS1_0"
      + name                             = "2i2ccarbonplanhubstorage"
      + nfsv3_enabled                    = false
      + primary_access_key               = (sensitive value)
      + primary_blob_connection_string   = (sensitive value)
      + primary_blob_endpoint            = (known after apply)
      + primary_blob_host                = (known after apply)
      + primary_connection_string        = (sensitive value)
      + primary_dfs_endpoint             = (known after apply)
      + primary_dfs_host                 = (known after apply)
      + primary_file_endpoint            = (known after apply)
      + primary_file_host                = (known after apply)
      + primary_location                 = (known after apply)
      + primary_queue_endpoint           = (known after apply)
      + primary_queue_host               = (known after apply)
      + primary_table_endpoint           = (known after apply)
      + primary_table_host               = (known after apply)
      + primary_web_endpoint             = (known after apply)
      + primary_web_host                 = (known after apply)
      + resource_group_name              = "2i2c-carbonplan-cluster"
      + secondary_access_key             = (sensitive value)
      + secondary_blob_connection_string = (sensitive value)
      + secondary_blob_endpoint          = (known after apply)
      + secondary_blob_host              = (known after apply)
      + secondary_connection_string      = (sensitive value)
      + secondary_dfs_endpoint           = (known after apply)
      + secondary_dfs_host               = (known after apply)
      + secondary_file_endpoint          = (known after apply)
      + secondary_file_host              = (known after apply)
      + secondary_location               = (known after apply)
      + secondary_queue_endpoint         = (known after apply)
      + secondary_queue_host             = (known after apply)
      + secondary_table_endpoint         = (known after apply)
      + secondary_table_host             = (known after apply)
      + secondary_web_endpoint           = (known after apply)
      + secondary_web_host               = (known after apply)
      + shared_access_key_enabled        = true

      + blob_properties {
          + change_feed_enabled      = (known after apply)
          + default_service_version  = (known after apply)
          + last_access_time_enabled = (known after apply)
          + versioning_enabled       = (known after apply)

          + container_delete_retention_policy {
              + days = (known after apply)
            }

          + cors_rule {
              + allowed_headers    = (known after apply)
              + allowed_methods    = (known after apply)
              + allowed_origins    = (known after apply)
              + exposed_headers    = (known after apply)
              + max_age_in_seconds = (known after apply)
            }

          + delete_retention_policy {
              + days = (known after apply)
            }
        }

      + identity {
          + identity_ids = (known after apply)
          + principal_id = (known after apply)
          + tenant_id    = (known after apply)
          + type         = (known after apply)
        }

      + network_rules {
          + bypass                     = (known after apply)
          + default_action             = (known after apply)
          + ip_rules                   = (known after apply)
          + virtual_network_subnet_ids = (known after apply)

          + private_link_access {
              + endpoint_resource_id = (known after apply)
              + endpoint_tenant_id   = (known after apply)
            }
        }

      + queue_properties {
          + cors_rule {
              + allowed_headers    = (known after apply)
              + allowed_methods    = (known after apply)
              + allowed_origins    = (known after apply)
              + exposed_headers    = (known after apply)
              + max_age_in_seconds = (known after apply)
            }

          + hour_metrics {
              + enabled               = (known after apply)
              + include_apis          = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
            }

          + logging {
              + delete                = (known after apply)
              + read                  = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
              + write                 = (known after apply)
            }

          + minute_metrics {
              + enabled               = (known after apply)
              + include_apis          = (known after apply)
              + retention_policy_days = (known after apply)
              + version               = (known after apply)
            }
        }

      + routing {
          + choice                      = (known after apply)
          + publish_internet_endpoints  = (known after apply)
          + publish_microsoft_endpoints = (known after apply)
        }

      + share_properties {
          + cors_rule {
              + allowed_headers    = (known after apply)
              + allowed_methods    = (known after apply)
              + allowed_origins    = (known after apply)
              + exposed_headers    = (known after apply)
              + max_age_in_seconds = (known after apply)
            }

          + retention_policy {
              + days = (known after apply)
            }

          + smb {
              + authentication_types            = (known after apply)
              + channel_encryption_type         = (known after apply)
              + kerberos_ticket_encryption_type = (known after apply)
              + versions                        = (known after apply)
            }
        }
    }

  # azurerm_storage_share.homes will be created
  + resource "azurerm_storage_share" "homes" {
      + enabled_protocol     = "SMB"
      + id                   = (known after apply)
      + metadata             = (known after apply)
      + name                 = "homes"
      + quota                = 100
      + resource_manager_id  = (known after apply)
      + storage_account_name = "2i2ccarbonplanhubstorage"
      + url                  = (known after apply)
    }

  # azurerm_subnet.node_subnet will be created
  + resource "azurerm_subnet" "node_subnet" {
      + address_prefix                                 = (known after apply)
      + address_prefixes                               = [
          + "10.1.0.0/16",
        ]
      + enforce_private_link_endpoint_network_policies = false
      + enforce_private_link_service_network_policies  = false
      + id                                             = (known after apply)
      + name                                           = "k8s-nodes-subnet"
      + resource_group_name                            = "2i2c-carbonplan-cluster"
      + virtual_network_name                           = "k8s-network"
    }

  # azurerm_virtual_network.jupyterhub will be created
  + resource "azurerm_virtual_network" "jupyterhub" {
      + address_space         = [
          + "10.0.0.0/8",
        ]
      + dns_servers           = (known after apply)
      + guid                  = (known after apply)
      + id                    = (known after apply)
      + location              = "westus2"
      + name                  = "k8s-network"
      + resource_group_name   = "2i2c-carbonplan-cluster"
      + subnet                = (known after apply)
      + vm_protection_enabled = false
    }

  # kubernetes_namespace.homes will be created
  + resource "kubernetes_namespace" "homes" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + name             = "azure-file"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
    }

  # kubernetes_secret.homes will be created
  + resource "kubernetes_secret" "homes" {
      + data = (sensitive value)
      + id   = (known after apply)
      + type = "Opaque"

      + metadata {
          + generation       = (known after apply)
          + name             = "access-credentials"
          + namespace        = "azure-file"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
    }

Plan: 15 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + kubeconfig            = (sensitive value)
  + registry_creds_config = (sensitive value)
```

</details>

### CODE REVIEW

**Comment by sgibson91** [comment]
@jhamman Would you like to check this and make sure naming conventions make sense, the machines look reasonable, etc?

File: terraform/azure/projects/carbonplan.tfvars [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    vm_size : "Standard_M64s_v2"
+  },
+  "vvhuge" : {
+    min : 0,
+    max : 20,
+    vm_size : "Standard_M128s_v2"
+  },
+}
+
+dask_nodes = {
```
Comment (yuvipanda) [inline terraform/azure/projects/carbonplan.tfvars:42]: In terraform, let's just leave this undefined - I think that should make it default to same set of sizes as notebook nodes.

**Comment by yuvipanda** [comment]
One comment, otherwise LGTM!

File: terraform/azure/projects/carbonplan.tfvars (hunk 1 above)
Comment (sgibson91) [reply terraform/azure/projects/carbonplan.tfvars:42]: I took my lead from the jsonnet file where there seem to be fewer dask nodes than notebook nodes? https://github.com/2i2c-org/infrastructure/blob/aaf5534a9fca24bc56cef9cae9ea3083d4f8cc2f/eksctl/carbonplan.jsonnet#L33-L38

Comment (yuvipanda) [reply terraform/azure/projects/carbonplan.tfvars:42]: Hmm, good point. I think keeping the notebook nodes match the dask nodes will make things easier to maintain for us. I'll work on fixing the jsonnet on eksctl to match that as well!

While looking through the code, I discovered that in our Azure terraform code, we don't actually default to dask nodes being same as notebook nodes! So I made https://github.com/2i2c-org/infrastructure/pull/834

Comment (sgibson91) [reply terraform/azure/projects/carbonplan.tfvars]: I removed the dask nodepools and updated against `master`. Will go for a deploy!

**Comment by sgibson91** [comment]
Thanks @yuvipanda and @jhamman! I will go for a deploy

**Review by yuvipanda — APPROVED** [review]

**Comment by sgibson91** [comment]
Not really sure why we're running into this if we're running k8s 1.20 as it seems to be a bug from moving to 1.16 from 1.15?

https://docs.microsoft.com/en-us/azure/aks/troubleshooting#why-do-upgrades-to-kubernetes-116-fail-when-using-node-labels-with-a-kubernetesio-prefix

https://github.com/2i2c-org/infrastructure/blob/831e8cc80f3629ec69bbab3b61c57b236844edc1/terraform/azure/variables.tf#L30-L39

**Comment by sgibson91** [comment]
Seems like it could be related to this: https://github.com/Azure/AKS/issues/1835

I'm not sure where to go from here, other than changing this line to something that doesn't have the `kubernetes.io` prefix:

https://github.com/2i2c-org/infrastructure/blob/bc0c7458248f52cd58751e72f33fb148e2d3756b/terraform/azure/main.tf#L137-L140

**Comment by yuvipanda** [comment]
@sgibson91 is this happening on cluster create? or was the cluster already created?

**Comment by sgibson91** [comment]
> @sgibson91 is this happening on cluster create? or was the cluster already created?

Cluster was already created, this was happening at nodepool creation time.

**Comment by yuvipanda** [comment]
I confirmed with @sgibson91 that this happens on the first run of terraform apply. As usual, things making not-sense on Azure :)

Let's use hub.jupyter.org/node-size or something similar? I'd love for us to not use something 2i2c specific.

**Comment by yuvipanda** [comment]
@sgibson91 alternatively, just try running terraform destroy and apply again?

**Comment by sgibson91** [comment]
@yuvipanda Was just in the middle of doing that!

**Comment by sgibson91** [comment]
Nope, destroy and reapply didn't work 😞 I'll try changing the label name

**Comment by sgibson91** [comment]
Using `hub.jupyter.org` instead of `node.kubernetes.io` worked. @yuvipanda I made this change in `main.tf` (see [`e8bf64f`](https://github.com/2i2c-org/infrastructure/pull/833/commits/e8bf64f3ee844cfa5eefd9cd5e068a0788f7c720)), was that sensible or did we want to be able to dynamically change labels from the `.tfvars` file?

**Comment by sgibson91** [comment]
I will merge this now and we can address the labelling in a new PR if needed

**Comment by yuvipanda** [comment]
@sgibson91 the change of label here will affect justice innovation lab as well I think. We'll need to do a terraform apply there, and then change [the hub config to use the new labels](https://github.com/2i2c-org/infrastructure/blob/da84ec39e603a801c0a5f7cf8605bba59500dec1/config/hubs/justiceinnovationlab.cluster.yaml#L49). Would you be able to take that on too?

**Comment by sgibson91** [comment]
@yuvipanda I can absolutely run tf apply and open the PR for the JIL hub config. The only thing I don't think I can to is actually run the deployer against JIL - IIRC I had issues with that when I submitted that PR that updated deprecated attributes in terraform.

Once I have the new carbon plan hub up, I'd like to spend some time automating our Azure-based deploys as that would 1) avoid this issue in the future, and ii) complete our trifecta of major cloud vendor deploys in CI/CD.

**Comment by sgibson91** [comment]
@yuvipanda Ah no, I can't do any of that. I don't have access to the JIL Azure subscription

```bash
$ tf plan -var-file=projects/justiceinnovationlab.tfvars -out=jil
╷
│ Error: building AzureRM Client: obtain subscription(xxxx-xxxx-xxxx-xxxx-xxxx) from Azure CLI: Error parsing json result from the Azure CLI: Error waiting for the Azure CLI: exit status 1: ERROR: Subscription 'xxxx-xxxx-xxxx-xxxx-xxxx' not found. Check the spelling and casing and try again.
│ 
│   with provider["registry.terraform.io/hashicorp/azurerm"],
│   on main.tf line 8, in provider "azurerm":
│    8: provider "azurerm" {
│ 
╵
```

**Comment by choldgraf** [comment]
Imo we should prioritize 

> Once I have the new carbon plan hub up, I'd like to spend some time automating our Azure-based deploys as that would 1) avoid this issue in the future, and 2) complete our trifecta of major cloud vendor deploys in CI/CD.

The JIL hub may actually wind down relatively soon so we shouldn't put a ton of time into improving it. Will update an issue w more info there

**Comment by sgibson91** [comment]
@choldgraf strong agree on your prioritisation there. Yuvi wasn't suggesting improving JIL really, just making sure it's running the most recent config that's on master.

**Comment by yuvipanda** [comment]
Yeah, agree 100% on improvements @choldgraf. But I don't want to leave it in an inconsistent state where changes to it aren't possible, as that tends to blow up. This is almost exactly what happened with the OpenScapes cluster - it being unmaintained meant that when we needed to touch it, it was in a bad state. 

@sgibson91 ah, yes I think I'm still the only one with Azure access to JIL :| I'll make a task to fix this tomorrow.

**Pull request merged**